pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -348,6 +348,20 @@ func (c *Controller) syncService(logger klog.Logger, key string) error {
 	}
 
 	if service.Spec.Selector == nil {
+		// Ensure existing endpoint slices managed by service are deleted when selector
+		// is made to be empty for the service.
+		esLabelSelector := labels.Set(map[string]string{
+			discovery.LabelServiceName: service.Name,
+			discovery.LabelManagedBy:   controllerName,
+		}).AsSelectorPreValidated()
+		endpointSlices, err := c.endpointSliceLister.EndpointSlices(service.Namespace).List(esLabelSelector)
+		if err != nil {
+			return err
+		}
+		err = c.reconciler.DeleteEndpointSlices(service, endpointSlices)
+		if err != nil {
+			return err
+		}
```

Comment:
This all seems very duplicated... I think it would be cleaner to remove the `if service.Spec.Selector == nil` check here and instead update `c.reconciler.Reconcile` to handle this case.

Reply:
why not remove the managed-by label only so we keep the existing behvior and the endpointslicemirroring controller can reconcile it?

Reply:
@aojea The endpointslice created by endpointslicemirroring controller is owned by endpoints whereas this object is owned by service, so doesn't it make sense to delete endpointslice here ?